contextualized/regression/tests.py

```diff
@@ -47,44 +48,122 @@ def setUp(self):
         self.c_dim, self.x_dim, self.y_dim = c_dim, x_dim, y_dim
         self.C, self.X, self.Y = C.numpy(), X.numpy(), Y.numpy()
 
-    def _quicktest(self, model, univariate=False, correlation=False, markov=False):
-        """
+    def _get_model(
+        self,
+        model,
+        dataclass_type,
+        pred_dl_type,
+        correlation=False,
+        univariate=False,
+        markov=False,
+    ):
 
-        :param model:
-        :param univariate:  (Default value = False)
-        :param correlation:  (Default value = False)
-        :param markov:  (Default value = False)
+        get_dataclass = {
+            "datamodule": lambda x, **kwargs: x.datamodule(**kwargs),
+            "dataloader": lambda x, **kwargs: x.dataloader(**kwargs),
+        }
 
-        """
-        print(f"\n{type(model)} quicktest")
+        # get subset of data to use for y_true
+        def get_xy_pred(dc):
```

Comment:
Tests should align with our usecases, and the datamodules main function is to hide all the individual datasets from the user. We should use the dataset kwarg in the trainer to specify prediction data, rather than adding functions like `get_xy_pred` and `_get_model`. I'd recommend deleting these and using the dataset kwarg in `_quicktest`.

Reply:
That makes sense. The main reason I created get_model was for handling the various y_true's (now dependent on predict type, dataclass type (dl, dm), and model type (correlation, etc.)). Would we want to add a `get_y_true` to each trainer along what you suggest? 

I just made a commit that hopefully can show what I mean (my changes to datasets.py weren't syncing before, but this should be updated too now!) At least with the tests we have, everything runs.

The only thing is I'm not sure is whether the y_true are calculated in a universal way. For instance, if I'm remembering, you had suggested changing y_true of correlation to  `y_true = np.tile(y_dset[:, :, np.newaxis], (1, 1, self.X.shape[-1] + self.Y.shape[-1]))`, but would this be the general behavior?